Assets/MixedRealityToolkit/_Core/Definitions/Utilities/InscribedRectangle.cs

```diff
@@ -0,0 +1,570 @@
+            Vector2 topRight = new Vector2(centerPoint.x + halfWidth, centerPoint.y + halfHeight);
+            Vector2 bottomLeft = new Vector2(centerPoint.x - halfWidth, centerPoint.y - halfHeight);
+            Vector2 bottomRight = new Vector2(centerPoint.x + halfWidth, centerPoint.y - halfHeight);
+
+            // Rotate the rectangle.
+            topLeft = RotatePoint(topLeft, centerPoint, angleRadians);
+            topRight = RotatePoint(topRight, centerPoint, angleRadians);
+            bottomLeft = RotatePoint(bottomLeft, centerPoint, angleRadians);
+            bottomRight = RotatePoint(bottomRight, centerPoint, angleRadians);
+
+            // Get the rectangle edges.
+            Edge topEdge = new Edge(topLeft, topRight);
+            Edge rightEdge = new Edge(topRight, bottomRight);
+            Edge bottomEdge = new Edge(bottomLeft, bottomRight);
+            Edge leftEdge = new Edge(topLeft, bottomLeft);
+
+            // Check for collisions with the boundary geometry. If any of our edges collide, 
+            // the rectangle will not fit within the playspace.
+            for (int i = 0; i < geometryEdges.Length; i++)
+            {
+                if (EdgeUtilities.IsValidPoint(EdgeUtilities.GetIntersectionPoint(geometryEdges[i], topEdge)) ||
+                    EdgeUtilities.IsValidPoint(EdgeUtilities.GetIntersectionPoint(geometryEdges[i], rightEdge)) ||
+                    EdgeUtilities.IsValidPoint(EdgeUtilities.GetIntersectionPoint(geometryEdges[i], bottomEdge)) ||
+                    EdgeUtilities.IsValidPoint(EdgeUtilities.GetIntersectionPoint(geometryEdges[i], leftEdge)))
+                {
+                    return false;
+                }
+            }
+
+            // No collsions found with the rectangle. Success!
+            return true;
+        }
+
+        /// <summary>
+        /// Attempt to fit the largest rectangle possible within the geometry.
+        /// </summary>
+        /// <param name="geometryEdges">The boundary geometry.</param>
+        /// <param name="centerPoint">The center point for the rectangle.</param>
+        /// <param name="angleRadians">The rotation, in radians, of the rectangle.</param>
+        /// <param name="minArea">The smallest allowed area.</param>
+        /// <param name="width">Returns the width of the rectangle.</param>
+        /// <param name="height">Returns the height of the rectangle.</param>
+        /// <returns>
+        /// True if a rectangle with an area greater than or equal to minArea was able to be fit
+        /// within the geometry at centerPoint.
+        /// </returns>
+        private bool TryFixMaximumRectangle(
+            Edge[] geometryEdges,
+            Vector2 centerPoint,
+            float angleRadians,
+            float minArea,
+            out float width,
+            out float height)
+        {
+            width = 0.0f;
+            height = 0.0f;
+
+            float[] aspectRatios = {
```

Comment:
Could be declared in the class scope so we don't have to allocate this every time we call this method.